app/main/util/date_validator.py

```diff
@@ -0,0 +1,334 @@
+from datetime import date, datetime
+
+from flask import current_app
+
+db_date_format = "%Y-%m-%d"
+python_date_format = "%d/%m/%Y"
+default_start_date = None
+check_future_date = True
+
+
+def validate_date_range(date_range):
+    date_from = None
+    date_to = None
+    if date_range is not None:
+        if "date_from" in date_range:
+            try:
+                dt_from = datetime.strptime(
+                    str(date_range["date_from"]), python_date_format
+                )
+                date_from = dt_from.strftime(db_date_format)
+            except ValueError:
+                current_app.logger.error(
+                    "Invalid [date from] value being passed in date range"
+                )
+
+        if "date_to" in date_range:
+            try:
+                dt_to = datetime.strptime(
+                    str(date_range["date_to"]), python_date_format
+                )
+                date_to = dt_to.strftime(db_date_format)
+            except ValueError:
+                current_app.logger.error(
+                    "Invalid [date to] value being passed in date range"
+                )
+
+    return {"date_from": date_from, "date_to": date_to}
+
+
+def get_date(day, month, year, value):
+    day_var = (
+        day.rjust(2, "0") if len(str(day)) == 1 else day if day else value * 2
+    )
+    month_var = (
+        month.rjust(2, "0")
+        if len(str(month)) == 1
+        else month if month else value * 2
+    )
+    year_var = year.ljust(4, "0") if year and len(str(year)) > 0 else value * 4
+
+    return [day_var, month_var, year_var]
+
+
+def generate_date_values(args):
+    date_from = get_date(
+        args.get("date_from_day"),
+        args.get("date_from_month"),
+        args.get("date_from_year"),
+        "0",
+    )
+
+    date_values = {
+        "1": int(date_from[0]) if date_from[0].isnumeric() else 0,
+        "2": int(date_from[1]) if date_from[1].isnumeric() else 0,
+        "3": int(date_from[2]) if date_from[2].isnumeric() else 0,
+    }
+
+    date_to = get_date(
+        args.get("date_to_day"),
+        args.get("date_to_month"),
+        args.get("date_to_year"),
+        "0",
+    )
+    date_values.update(
+        {
+            "4": int(date_to[0]) if date_from[0].isnumeric() else 0,
+            "5": int(date_to[1]) if date_from[1].isnumeric() else 0,
+            "6": int(date_to[2]) if date_from[2].isnumeric() else 0,
+        }
+    )
+
+    return date_values
+
+
+def generate_default_date_values(args, start_date=None):
+    global default_start_date
```

Comment:
Do we need to set default_start_date as a global variable? I'd suggest avoiding the use of global variables unless a really good reason for it, since it makes things much harder to debug and maintain